core/sql/optimizer/RelGrby.h

```diff
@@ -552,7 +560,9 @@ class GroupByAgg : public RelExpr
   // ---------------------------------------------------------------------
   ItemExpr    * groupExprTree_;
   ValueIdSet  groupExpr_;
-
+  
+  ItemExpr  * extraGrpOrderby_;
+  ValueIdList  extraOrderExpr_;
```

Comment:
Anoop added the rollup feature recently, see https://github.com/apache/incubator-trafodion/pull/733. This also introduces an extra order by expression for a groupby. Maybe you and Anoop can decide whether you want to combine your code, for example by using this extra order for the ROLLUP groupby as well.